Microsoft.Toolkit.Uwp.UI.Controls/RadialProgressBar/RadialProgressBar.cs

```diff
@@ -0,0 +1,201 @@
+        }
+
+        /// <summary>
+        /// Called when the Minimum property changes.
+        /// </summary>
+        /// <param name="oldMinimum">Old value of the Minimum property.</param>
+        /// <param name="newMinimum">New value of the Minimum property.</param>
+        protected override void OnMinimumChanged(double oldMinimum, double newMinimum)
+        {
+            base.OnMinimumChanged(oldMinimum, newMinimum);
+            RenderSegment();
+        }
+
+        /// <summary>
+        /// Called when the Maximum property changes.
+        /// </summary>
+        /// <param name="oldMaximum">Old value of the Maximum property.</param>
+        /// <param name="newMaximum">New value of the Maximum property.</param>
+        protected override void OnMaximumChanged(double oldMaximum, double newMaximum)
+        {
+            base.OnMaximumChanged(oldMaximum, newMaximum);
+            RenderSegment();
+        }
+
+        /// <summary>
+        /// Called when the Value property changes.
+        /// </summary>
+        /// <param name="oldValue">Old value of the Value property.</param>
+        /// <param name="newValue">New value of the Value property.</param>
+        protected override void OnValueChanged(double oldValue, double newValue)
+        {
+            base.OnValueChanged(oldValue, newValue);
+            RenderSegment();
+        }
+
+        /// <summary>
+        /// Update the visual state of the control when its template is changed.
+        /// </summary>
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            OutlineFigure = GetTemplateChild(OutlineFigurePartName) as PathFigure;
+            OutlineArc = GetTemplateChild(OutlineArcPartName) as ArcSegment;
+            BarFigure = GetTemplateChild(BarFigurePartName) as PathFigure;
+            BarArc = GetTemplateChild(BarArcPartName) as ArcSegment;
+
+            RenderAll();
+        }
+
+        /// <summary>
+        /// Gets or sets the thickness of the circular ouline and segment
+        /// </summary>
+        public double Thickness
+        {
+            get { return (double)GetValue(ThicknessProperty); }
+            set { SetValue(ThicknessProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the Thickness dependency property
+        /// </summary>
+        public static readonly DependencyProperty ThicknessProperty = DependencyProperty.Register("Thickness", typeof(double), typeof(RadialProgressBar), new PropertyMetadata(4.0, ThicknessChangedHandler));
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RadialProgressBar"/> class.
+        /// Create a default circular progress bar
+        /// </summary>
+        public RadialProgressBar()
+        {
+            this.DefaultStyleKey = typeof(RadialProgressBar);
+
+            Foreground = Application.Current.Resources[DefaultForegroundColorBrushName] as SolidColorBrush;
```

Comment:
These should not be set in code - should be set in your generic XAML definition so that it can be properly styled. For example:

`<Setter Property="Foreground" Value="{StaticResourse SystemControlHighlightAccentBrush}" /> `